tests/lib/nrf_modem_lib/trace_backends/rtt/src/main.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <string.h>
+
+#include <zephyr/kernel.h>
+
+#include <unity.h>
+
+#include "trace_backend.h"
+
+#include "mock_SEGGER_RTT.h"
+
+extern int unity_main(void);
+
+static int trace_rtt_channel;
+
+/* Suite teardown shall finalize with mandatory call to generic_suiteTearDown. */
+extern int generic_suiteTearDown(int num_failures);
+
+void setUp(void)
+{
+	mock_SEGGER_RTT_Init();
+}
+
+void tearDown(void)
+{
+	mock_SEGGER_RTT_Verify();
+}
+
+int test_suiteTearDown(int num_failures)
+{
+	return generic_suiteTearDown(num_failures);
+}
+
+static int rtt_allocupbuffer_callback(const char *sName, void *pBuffer, unsigned int BufferSize,
+				      unsigned int Flags, int no_of_calls)
+{
+	char *exp_sName = "modem_trace";
+
+	TEST_ASSERT_EQUAL_CHAR_ARRAY(exp_sName, sName, sizeof(exp_sName));
+	TEST_ASSERT_NOT_EQUAL(NULL, pBuffer);
+	TEST_ASSERT_EQUAL(CONFIG_NRF_MODEM_LIB_TRACE_BACKEND_RTT_BUF_SIZE, BufferSize);
+	TEST_ASSERT_EQUAL(SEGGER_RTT_MODE_BLOCK_IF_FIFO_FULL, Flags);
+
+	return trace_rtt_channel;
+}
+
+static void trace_session_setup_with_rtt_transport(void)
+{
+	__wrap_SEGGER_RTT_AllocUpBuffer_ExpectAnyArgsAndReturn(trace_rtt_channel);
+	__wrap_SEGGER_RTT_AllocUpBuffer_AddCallback(&rtt_allocupbuffer_callback);
+	TEST_ASSERT_EQUAL(0, trace_backend_init());
+}
+
+static void expect_rtt_write(const uint8_t *buffer, uint16_t size)
+{
+	__wrap_SEGGER_RTT_Write_ExpectAndReturn(trace_rtt_channel, buffer, size, size);
+}
+
+void test_trace_backend_init_rtt(void)
+{
+	trace_rtt_channel = 1;
+	trace_session_setup_with_rtt_transport();
+}
```

Comment:
nit: Can we change this so that the function we are testing is visible in the test code? The whole body of this function is `trace_session_setup_with_rtt_transport()`, I would just copy paste it here, and wherever we want to run it again we can call this function `test_trace_backend_init_rtt()` instead.